misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.datamapper;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.VariableSymbol;
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.SpreadFieldNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Visitor to extract Record Type Structure information.
+ */
+public class DataMapperNodeVisitor extends NodeVisitor {
+    public final HashMap<String, String> restFields;
+    public final HashMap<String, Map<String, RecordFieldSymbol>> spreadFields;
+    public final ArrayList<String> specificFieldList;
+    private String rhsSymbolName;
+    private SemanticModel model;
+
+    public DataMapperNodeVisitor(String rhsSymbolName) {
+        this.restFields = new HashMap<>();
+        this.spreadFields = new HashMap<>();
+        this.rhsSymbolName = rhsSymbolName;
+        this.specificFieldList = new ArrayList<>();
+    }
+
+    public void setModel(SemanticModel model) {
+        this.model = model;
+    }
+
+    @Override
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
+        if (variableDeclarationNode.initializer().isPresent()) {
+            if (variableDeclarationNode.initializer().get().kind() == SyntaxKind.MAPPING_CONSTRUCTOR) {
+                String rightSymbolName = variableDeclarationNode.typedBindingPattern().typeDescriptor().toString().
```

Comment:
Why do we use the string representation of the type descriptors and use `replaceAll` APIs here? We need to find an alternative approach to deal with this usage such as using the type descriptor's kind and etc

Reply:
Modified the code according to the comments